### PR TITLE
grandorgue: init at 0.3.1-r2333

### PR DESCRIPTION
**Pull request by puzzlewolf**
###### Motivation for this change
Create a package for grandorgue, a sample based pipe organ simulator.
https://sourceforge.net/projects/ourorgan/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/grandorgue/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  pname = "grandorgue";
+  rev = "2333";
+  version = "0.3.1-r${rev}";
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/ourorgan/svn/trunk";
+    inherit rev;
+    sha256 = "0xzjdc2g4gja2lpmn21xhdskv43qpbpzkbb05jfqv6ma2zwffzz1";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config gcc ];
```
Comment (symphorien) [inline pkgs/applications/audio/grandorgue/default.nix]: I think `gcc` is already provided by `stdenv`.

**Review by symphorien — CHANGES_REQUESTED** [review]
Thanks for the PR.
The package looks good, except for one nitpick.
I built grandOrgue with nix-review and it seems to work (makes organ-like sound), except that I get this scary warning at startup:
```
RtMidi error: MidiInAlsa::initialize: error creating ALSA sequencer client object.
RtMidi error: MidiOutAlsa::initialize: error creating ALSA sequencer client object.
```
Not sure if it is normal.
I am also intrigued at how the open file dialog works even without `wrapGAppsHook`, but since it works, let's not change anything in this regard.

File: pkgs/applications/audio/grandorgue/default.nix (hunk 1 above)
Comment (puzzlewolf) [reply pkgs/applications/audio/grandorgue/default.nix]: Thanks for the feedback, I removed `gcc`.

Concerning the warning: I don't see that warning on my NixOS machine, but on ArchLinux (on different hardware), it appears too. I doesn't seem to have any consequences, the program works just fine. I think I'll just use the package, and try to fix problems as they come along :-)

**Comment by puzzlewolf** [comment]
I found an answer to the RtMidi error at https://sourceforge.net/p/ourorgan/discussion/962124/thread/9875b915/: it should not occur if the user is a member of the `audio` group. However, the program works even with the error. So, I think this package should be fine :)

**Review by symphorien — APPROVED** [review]
adding myself to the audio group works. Thanks!

**Review by Valodim — APPROVED** [review]
works for me.

not sure we can do anything about the `audio` group thing at package level. there is [general advice](https://nixos.wiki/wiki/PulseAudio) to that effect already, with reasonable discoverability.

File: pkgs/applications/audio/grandorgue/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  cmakeFlags = lib.optional (!jackaudioSupport) [
+    "-DRTAUDIO_USE_JACK=OFF"
+    "-DRTMIDI_USE_JACK=OFF"
+  ] ++ lib.optional (!includeDemo) "-DINSTALL_DEMO=OFF";
+
+  meta = {
+    description = "Virtual Pipe Organ Software";
+    homepage = "https://sourceforge.net/projects/ourorgan";
+    license = stdenv.lib.licenses.gpl2Plus;
```
Comment (grahamc) [inline pkgs/applications/audio/grandorgue/default.nix]: It looks like it is licensed gpl2 only, not gpl2Plus?

**Comment by grahamc** [comment]
Hmm....

```
$ ./result/bin/GrandOrgue
ALSA lib pcm.c:2635:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2635:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2635:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:869:(find_matching_chmap) Found no matching channel map
ALSA lib pcm.c:2556:(snd_pcm_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_pcm_pulse.so cannot be opened or _snd_pcm_pulse_open was not defined inside
ALSA lib pcm.c:2556:(snd_pcm_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_pcm_pulse.so cannot be opened or _snd_pcm_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
ALSA lib control.c:1394:(snd_ctl_open_conf) Either /nix/store/5g0ynqislkmbjr0q9zn2pp6g0qw1wg9l-alsa-plugins-1.2.1/lib/alsa-lib/libasound_module_ctl_pulse.so cannot be opened or _snd_ctl_pulse_open was not defined inside
zsh: segmentation fault (core dumped)  ./result/bin/GrandOrgue
```

I don't really know what to make of this.

**Comment by grahamc** [comment]
The `segfault` was `myfault`. I think the only trouble is the license?

File: pkgs/applications/audio/grandorgue/default.nix (hunk 2 above)
Comment (puzzlewolf) [reply pkgs/applications/audio/grandorgue/default.nix]: Don't know where I got the 2 plus, fixing it right now. Thanks for reviewing :)

**Comment by puzzlewolf** [comment]
Thanks for finding that license mistake :+1:

**Comment by grahamc** [comment]
Thank you!

**Pull request merged**